### PR TITLE
Refactored ViewportQuadSpec

### PR DESCRIPTION
**Pull request by lilleyse**
For #1259 

There's an issue where `ClearCommand.ALL.execute(context)` doesn't seem to do anything, and instead the clear color ends up being the scene's background color (0,0,0,255).  I noticed other specs account for this, like [TextureAtlasSpec](https://github.com/AnalyticalGraphicsInc/cesium/blob/9159c4c5a5d2340d2b6fbdc34504bdc01ea213e2/Specs/Scene/TextureAtlasSpec.js#L119-L120).

I'm going to look into this more, but decided to start the pull request anyways.

### CODE REVIEW

File: Specs/Scene/ViewportQuadSpec.js [hunk 1]
```diff
@@ -20,44 +17,34 @@ defineSuite([
...
+        scene.destroyForSpecs();
     });
 
     beforeEach(function() {
         viewportQuad = new ViewportQuad();
         viewportQuad.rectangle = new BoundingRectangle(0, 0, 2, 2);
-
-        us = context.uniformState;
-        us.update(context, createFrameState(createCamera()));
+        scene.primitives.add(viewportQuad);
```
Comment (pjcozzi) [inline Specs/Scene/ViewportQuadSpec.js]: I suggest writing
```
viewportQuad = new ViewportQuad();
scene.primitives.add(viewportQuad);
```
as
```
viewportQuad = scene.primitives.add(new ViewportQuad());
```
This code was written before `add` returned the added primitive.

**Comment by pjcozzi** [comment]
> There's an issue where `ClearCommand.ALL.execute(context)`...

Don't use `ClearCommand` directly since the test is built on `Scene` now; let `Scene` execute all of the commands.

Instead of adding the primitive in `beforeEach`, we can test like:
```javascript
expect(scene.renderForSpecs()).toEqual([0, 0, 0, 255]);
scene.primitives.add(viewportQuad);
expect(scene.renderForSpecs()).not.toEqual([0, 0, 0, 255]);
```
which, of course, makes my tip above not useful.

**Comment by lilleyse** [comment]
Updated.

File: Specs/Scene/ViewportQuadSpec.js [hunk 2]
```diff
@@ -77,42 +63,40 @@ defineSuite([
             new Color(1.0, 1.0, 1.0, 1.0));
     });
 
-    it('throws when rendered with without a rectangle', function() {
+    it('throws when rendered without a rectangle', function() {
         viewportQuad.rectangle = undefined;
 
         expect(function() {
-            render(context, frameState, viewportQuad);
+            scene.primitives.add(viewportQuad);
```
Comment (pjcozzi) [inline Specs/Scene/ViewportQuadSpec.js]: Only put the part that we expect to throw in the callback so, here and below, move this above:
```javascript
scene.primitives.add(viewportQuad);
```

**Comment by pjcozzi** [comment]
Just that minor comment.

**Comment by lilleyse** [comment]
Done

**Comment by pjcozzi** [comment]
Breathtaking.

**Pull request merged**